File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        s = set()
        L = 0

        for R in range(len(nums)):
            if R - L > k:
                s.remove(nums[L])
                L += 1 
            if nums[R] in s:
                return True
            s.add(nums[R])
        return False

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set = new HashSet<Character>();
        int res = 0;

        for(int l = 0, r = 0; r < s.length(); r++){
            Character curChar = s.charAt(r);

            while(set.contains(curChar)){
                set.remove(s.charAt(l));
                l++;
            }

            res = Math.max(res, r - l + 1);
            set.add(curChar);
        }

        return res;
    }
}
